src/modules/organisations/organisations.controller.ts

```diff
@@ -154,4 +156,51 @@ export class OrganisationsController {
   async getById(@Param('org_id') org_id: string) {
     return this.organisationsService.getOrganizationDetailsById(org_id);
   }
+
+  @ApiOperation({ summary: 'Export members of an Organisation to a CSV file' })
+  @ApiResponse({
+    status: 200,
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'The CSV file containing organisation members is returned.',
+    headers: {
+      'Content-Type': {
+        description: 'The content type of the response, which is text/csv.',
+        schema: {
+          type: 'string',
+          example: 'text/csv',
+        },
+      },
+      'Content-Disposition': {
+        description: 'Indicates that the content is an attachment with a filename.',
+        schema: {
+          type: 'string',
+          example: 'attachment; filename="organisation-members-{orgId}.csv"',
+        },
+      },
+    },
+  })
+  @UseGuards(OwnershipGuard)
+  @Get(':org_id/members/export')
+  async exportOrganisationMembers(@Param('org_id') orgId: string, @Req() req: Request, @Res() res: Response) {
+    const userId = req['user'].id;
+    const filePath = await this.organisationsService.exportOrganisationMembers(orgId, userId);
+    const fileStream = createReadStream(filePath);
+
+    res.set({
+      'Content-Type': 'text/csv',
+      'Content-Disposition': `attachment; filename="organisation-members-${orgId}.csv"`,
+    });
+
+    fileStream.pipe(res);
```

Comment:
Better to use pipeline to stream the files it offers better internal buffer management and reduces drainage. Your implementation is good, but pipeline is better for production